Revision.md

```diff
@@ -0,0 +1,238 @@
+
+# Revision
+
+Following is an [IPLD Schema] representation of the `Revision` objects (or whatever we want to call it), which:
+
+1. Can be in `Draft` or `Release` state.
+    - Allow _coordinated_ `Release` updates.
+    - Allow _concurrent_ `Draft` updates.
+3. Are identified by an [ed25519][] public key, that can represent
+    - [IPNS][] names.
+    - [did:key][] identifiers.
+    - Actors in [UCAN][] authorization.
+
+
+This design would address several problems in .storage services:
+
+### Large uploads
+
+Large uploads that span multiple CAR files would gain a first class representation via `Revision` objects. Client application wishing to upload large file (or any other DAG) will be able to accomplish that by:
+
+
+1. Self issuing a new `Revision` identifier (and corresponding UCAN) by generating new [ed25519][] keypair.
+1. Submitting concurrent `Patch` transactions (in [CAR][] format). Each transaction will contain DAG shards, subset of blocks that would feet upload quota.
+1. Finalizing upload by submitting `Commit` transaction (in [CAR][] format), setting `root` of the `Revision` to a `CID` of the large file.
+
+
+This would allow .storage service to list "in progress" uploads (keyed by `Revision` id) and "finished" uploads (keyed by `CID` or/and `Revision` id).
+
+### IPNS
+
+.storage services could mirror `Revisions`s to corresponding [IPNS][] names, making it possible to access arbitrary uploads / pins through an IPNS resoultion.
+
+> `Revision` state (`Draft` or `Release`) could be used to decide when to propagate changes through the network e.g. sevice could choose to only announce only `Release` states.
+
+
+### did:key
+
+.storage service could also provide interface for accessing content under `did:key` that correspond to a given keys. Basically we can build IPNS like system except with delegated publishing through UCANs before integrating that into IPNS.
+
+### UCAN
+
+By representing `Revision`s as first class objects identified by `did:key` they become actors in UCANs delegated capabilties system.
+
+.storage user could issue delegated token for specific `Revision` object and excercise that capability to update given `Revision` object or delegate that capability to another actor in the system.
+
+
+## Schema
+
+Following is an [IPLD schema][] definition for the `Revision` object. 
+
+
+```ipldsch
+-- Revision represents (IPNS) named pointer to a DAG that
+-- is either in "draft" or fully "release" state.
+-- In both cases it is anotated pointer to a DAG.
+type Pin union {
```

Comment:
i think you already know this, but let’s get rid of the work “Pin” anywhere in the document :)

for this structure specifically, i think you can call it “Session”